### DISCUSSION TITLE
PV + Offpeak

### DISCUSSION THREAD

**Discussion by OttoKorrect** [pkuehnel/TeslaSolarCharger#1065] [General]
I have offpeak between 00:00 and 06:00 everyday from my electricity provider.
What I would want is that _Min SOC_ is only applied in this time slot, and that at all other times it is PV Only.

I thought that _Spot Price + PV_ and scheduled everyday for 6:00 might do that, but it will try to charge outside of the off-peak period.
I have set Fixed Price 00:00-06:00 that is 1/10th of the base price.
eg here it will charge at 13:00 with no regard of the fixed price or enough solar
![image](https://github.com/pkuehnel/TeslaSolarCharger/assets/7450377/b1e22eea-de55-4510-a7e0-9070fa834b99)

Have I misunderstood something, in regards to charge mode? Or TSC can't do what I want at this time?

**Comment by pkuehnel** [comment]
Spot Price charge mode is currently fixed to the German Spot price market (different prices for every hour every day). Planning charges based on time based prices is currently not implemented as there is more to come at this end. But there is a Workaround: Set Charge Mode to PV Only, Min SOC to 60, enable charge everyday and set the time to 6:00. Note: I highly recommend to enable `Autoset charge starts` in the car config. This sets a scheduled charge start in the Tesla App, resulting in the car starting to charge on time even if there are issues with the Tesla API (e.g. car has no internet connection,...).

**Reply by OttoKorrect** [reply]
Ok I will give that a go - again thanks for being so responsive to questions and also making updates to improve issues when you find them.
